src/NHibernate/Intercept/DefaultDynamicLazyFieldInterceptor.cs

```diff
@@ -51,16 +51,7 @@ public object Intercept(InvocationInfo info)
 				}
 			}
 
-			object returnValue;
-			try
-			{
-				returnValue = info.InvokeMethodOnTarget();
-			}
-			catch (TargetInvocationException ex)
-			{
-				throw ReflectHelper.UnwrapTargetInvocationException(ex);
-			}
-			return returnValue;
+			return info.InvokeMethodOnTarget();
```

Comment:
Unwrapping moved into `InvokeMethodOnTarget`, which is called two times in NHibernate code: here and a couple of lines above here.